storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -138,7 +129,11 @@ func (s *fileSystemSource) enqueueReaders(
 
 		readers := make([]fs.FileSetReader, 0, len(files))
 		for i := 0; i < len(files); i++ {
-			r := readerPool.get()
+			r, err := readerPool.get()
+			if err != nil {
+				s.log.Errorf("unable to get reader from pool")
```

Comment:
should this be bubbled up so we know that we've bootstrapped from partial data.

Reply:
Sure thing, restructured this to bubble it up.